Storage/StorageTests/Tools/StorageTypeDeletionsTests.swift

```diff
@@ -0,0 +1,62 @@
+import XCTest
+
+@testable import Storage
+
+final class StorageTypeDeletionsTests: XCTestCase {
+
+    private let sampleSiteID: Int64 = 98765
+
+    private var storageManager: StorageManagerType!
+
+    private var storage: StorageType! {
+        storageManager.viewStorage
+    }
+
+    override func setUp() {
+        super.setUp()
+        storageManager = CoreDataManager(name: "WooCommerce", crashLogger: MockCrashLogger())
+    }
+
+    override func tearDown() {
+        storageManager.reset()
+        storageManager = nil
+        super.tearDown()
+    }
+
+    func test_deleteCoupon_by_siteID_deletes_all_coupons() throws {
+        // Given
+        let coupon1 = storage.insertNewObject(ofType: Coupon.self)
+        coupon1.siteID = sampleSiteID
+        coupon1.couponID = 1
+
+        let coupon2 = storage.insertNewObject(ofType: Coupon.self)
+        coupon2.siteID = sampleSiteID
+        coupon2.couponID = 2
```

Comment:
I think that here you should check that first the two objects exist in the Storage after adding them, otherwise the test, also if you delete the coupons, will be always green.